src/services/RPCSelector.ts

```diff
@@ -0,0 +1,69 @@
+import { JsonRpcProvider, Block } from '@ethersproject/providers';
+import { isTimestampMoreRecentThan } from '../utils/helpers';
+
+interface ProviderComparand {
+  blockNumber: number;
+  url: string;
+}
+
+class RPCSelector {
+  readonly urls: string[];
+  readonly preferredProviderUrl: string;
+  readonly rpcRequestTimeout: number;
+  readonly maxBlockAge: number;
+
+  constructor(urls: string | string[], rpcRequestTimeout = 15000, maxBlockAge = 60) {
+    this.urls = typeof urls === 'string' ? urls.split(',') : urls;
```

Comment:
urls is supposed to be string. Why do you expect it be a non-string type?

Reply:
I wanted to add support to both string and string array arguments. Devs might want to instantiate it with an array already. Besides that, there's no other real reason. The behavior will be the same.